src/backend/partaj/core/views/common.py

```diff
@@ -1,15 +1,114 @@
 """
 Common views that serve a purpose for any Partaj user.
 """
+import csv
 import mimetypes
 
 from django.contrib.auth.mixins import LoginRequiredMixin
+from django.db.models import DateTimeField, Exists, ExpressionWrapper, F, OuterRef, Q
 from django.http import FileResponse, HttpResponse
 from django.shortcuts import redirect
 from django.views import View
 from django.views.generic import TemplateView
 
-from ..models import ReferralAnswerAttachment, ReferralAttachment
+from .. import models
+from ..models import ReferralAnswerAttachment, ReferralAttachment, ReferralState
+
+
+class Export(LoginRequiredMixin, View):
+    """
+    Return all referral in csv files to authenticated users.
+    """
+
+    def get_queryset(self):
+        """
+        filter referral and  return a ready-to-use queryset.
+        """
+
+        queryset = (
+            models.Referral.objects.exclude(state=ReferralState.DRAFT)
+            .annotate(
+                due_date=ExpressionWrapper(
+                    F("sent_at") + F("urgency_level__duration"),
+                    output_field=DateTimeField(),
+                )
+            )
+            .annotate(unit=F("topic__unit__id"))
```

Comment:
Maybe we should annotate the name of the unit instead? End users wouldn't recognize Partaj IDs